rspace/src/main/scala/coop/rchain/rspace/merger/StateChange.scala

```diff
@@ -0,0 +1,159 @@
+package coop.rchain.rspace.merger
+
+import cats.Monoid
+import cats.effect.concurrent.Ref
+import cats.effect.{Concurrent, Sync}
+import cats.syntax.all._
+import coop.rchain.rspace.channelStore.ChannelStore
+import coop.rchain.rspace.hashing.{Blake2b256Hash, StableHashProvider}
+import coop.rchain.rspace.history.HistoryReaderBinary
+import coop.rchain.rspace.merger.MergingLogic.{consumesAffected, producesAffected}
+import coop.rchain.rspace.serializers.ScodecSerialize.{JoinsB, WaitingContinuationB}
+import coop.rchain.rspace.syntax._
+import coop.rchain.shared.Serialize
+import coop.rchain.shared.syntax._
+import fs2.Stream
+import scodec.bits.ByteVector
+
+/** Waiting continuation (WK) with channels of consume that put this WK. This class is used to compute joins changes. */
+final case class ConsumeChange(body: ByteVector, channels: Seq[Blake2b256Hash])
+
+/** We need this index to compute joins changes base on consume changes. This is because joins store not hashes of
+  * channels but channels themselves. So when we have channels from consumes (which are hashes of channels) -
+  * we are not able to reconstruct full join ByteVector */
+final case class JoinIndex(body: ByteVector, channels: Seq[Blake2b256Hash])
```

Comment:
This index is required as joins store the whole joins not just hashes of channels